TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/Drivebase.java

```diff
@@ -0,0 +1,275 @@
+     * @param xInput    Strafing input. Negative is left, positive is right. [-1, 1].
+     * @param turnInput Turning input. Negative is ccw, positive is clockwise. [-1, 1].
+     */
+    public @Api void mecanumDrive(double yInput, double xInput, double turnInput) {
+        double frontLeft = yInput + xInput + turnInput;
+        double frontRight = yInput - xInput - turnInput;
+        double backLeft = yInput - xInput + turnInput;
+        double backRight = yInput + xInput - turnInput;
+
+        double clamp = maxOf(1, frontLeft, frontRight, backLeft, backRight);
+
+        fldrive.setPower(frontLeft / clamp);
+        frdrive.setPower(frontRight / clamp);
+        bldrive.setPower(backLeft / clamp);
+        brdrive.setPower(backRight / clamp);
+    }
+
+
+    /**
+     * For use in auto. Drives forward a specified distance, using the encoders.
+     *
+     * @param inches    The amount of inches to drive forward, negative is backwards.
+     * @param power     How fast to drive. (0, 1].
+     * @param telemetry Pass telemetry if you want this method to log to telemetry.
+     * @see Drivebase#driveSideways
+     * @see Drivebase#turnAngle
+     * @see Drivebase#turnToAngle
+     */
+    public @Api void driveForward(double inches, double power, @Nullable Telemetry telemetry) {
+        setMotorModes(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+
+        double target = inches * ENCODER_PER_INCH;
+
+        fldrive.setTargetPosition((int) target);
+        frdrive.setTargetPosition((int) target);
+        bldrive.setTargetPosition((int) target);
+        brdrive.setTargetPosition((int) target);
+
+        fldrive.setPower(power);
+        frdrive.setPower(power);
+        bldrive.setPower(power);
+        brdrive.setPower(power);
+
+        setMotorModes(DcMotor.RunMode.RUN_TO_POSITION);
+
+        waitForMotors(telemetry);
+
+        fldrive.setPower(0);
+        frdrive.setPower(0);
+        bldrive.setPower(0);
+        brdrive.setPower(0);
+    }
+
+    /**
+     * For use in auto. Strafes sideways a specified amount of inches, using encoders.
+     *
+     * @param inches    How many inches to strafe, negative is left.
+     * @param power     How fast to strafe. (0, 1].
+     * @param telemetry Pass this if you want the method to log to telemetry.
+     * @see Drivebase#driveForward
+     * @see Drivebase#turnAngle
+     * @see Drivebase#turnToAngle
+     */
+    public @Api void driveSideways(double inches, double power, @Nullable Telemetry telemetry) {
+        setMotorModes(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+
+        double target = inches * ENCODER_PER_INCH;
+
+        fldrive.setTargetPosition((int) target);
+        frdrive.setTargetPosition((int) -target);
+        bldrive.setTargetPosition((int) -target);
+        brdrive.setTargetPosition((int) target);
+
+        fldrive.setPower(power);
+        frdrive.setPower(-power);
+        bldrive.setPower(-power);
```

Comment:
When using run to position, you should never use a negative power, **only** a negative distance.

Setting power should also be after setting the RunMode to RUN_TO_POSITION